django/utils/asyncio.py

```diff
@@ -31,3 +31,14 @@ def inner(*args, **kwargs):
         return decorator(func)
     else:
         return decorator
+
+
+async def alist(async_iterable):
+    """
+    Turns an async generator into a list (since list() expects a synchronous
+    generator)
+    """
+    result = []
+    async for item in async_iterable:
+        result.append(item)
+    return result
```

Comment:
Unfamiliar with the nuances of asyncio and whether it would be valid, but is it _possible_ to turn this into an async list comprehension?

Reply:
Oh almost certainly, I just try to avoid list comprehensions where I can to improve readability.

Reply:
Not sure there's a lot of value in proving this utility tbh. I would assume users familiar with async are already aware of this pitfall and others will have a hard time discovering this undocumented utility when getting an exception when trying to do sync work in an async context?

I could see us adjust `QuerySet.__iter__` to catch such exception and hint at using this pattern but in its current state (no proper docs, only an inline mention of the function) I don't think it's bringing enough value to warrant an inclusion in core.